minimap2.wdl

```diff
@@ -0,0 +1,142 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task Indexing {
+    input {
+        File referenceFile
+        String outputPrefix
+        Boolean useHomopolymerCompressedKmer = false
+        Int kmerSize = 15
+        Int minimizerWindowSize = 10
+
+        Int? splitIndex
+
+        Int cores = 1
+        Int memory = 4
+        String dockerImage = "quay.io/biocontainers/minimap2:2.17--h84994c4_0"
+    }
+
+    command {
+        set -e pipefail
+        mkdir -p $(dirname ~{outputPrefix})
+        minimap2 \
+        ~{true="-H" false="" useHomopolymerCompressedKmer} \
+        ~{"-k " + kmerSize} \
+        ~{"-w " + minimizerWindowSize} \
+        ~{"-I " + splitIndex} \
+        ~{"-d " + outputPrefix + ".mmi"} \
+        ~{"-t " + cores} \
+        ~{referenceFile}
+    }
+
+    output {
+        File outputIndexFile = outputPrefix + ".mmi"
+    }
+
+    runtime {
+        cpu: cores
+        memory: memory
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        referenceFile: "Reference fasta file."
+        outputPrefix: "Output directory path + output file prefix."
+        useHomopolymerCompressedKmer: "Use homopolymer-compressed k-mer (preferrable for PacBio)."
+        kmerSize: "K-mer size (no larger than 28)."
+        minimizerWindowSize: "Minimizer window size."
+        splitIndex: "Split index for every ~NUM input bases."
+
+        outputIndexFile: "Indexed reference file."
+    }
+}
+
+task Mapping {
+    input {
+        File queryFile
+        File referenceFile
+        String outputPrefix
+        String presetOption
+        Boolean outputSAM = false
+
+        Int? maxFragmentLength
+        Int? maxIntronLength
+        Boolean? skipSelfAndDualMappings
+        Int? retainMaxSecondaryAlignments
+        Int? matchingScore
+        Int? mismatchPenalty
+        String? howToFindGTAG
+        Boolean? secondaryAlignment
+
+        Int cores = 4
+        Int memory = 7
+        String dockerImage = "quay.io/biocontainers/minimap2:2.17--h84994c4_0"
+    }
+
+    command {
+        set -e pipefail
+        mkdir -p $(dirname ~{outputPrefix})
+        minimap2 \
+        ~{"-x " + presetOption} \
+        ~{true="-a" false="" outputSAM} \
+        ~{"-G " + maxIntronLength} \
+        ~{"-F " + maxFragmentLength} \
+        ~{true="-X" false="" skipSelfAndDualMappings} \
+        ~{"-N " + retainMaxSecondaryAlignments} \
+        ~{"-A " + matchingScore} \
+        ~{"-B " + mismatchPenalty} \
+        ~{"-u " + howToFindGTAG} \
+        ~{true="--secondary=yes" false="--secondary=no" secondaryAlignment} \
```

Comment:
`--secondary=` could be outside of the `~{}`, that would mean less bytes :stuck_out_tongue: